src/containers/org/components/DeleteOrgModal.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * @flow
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { ActionModal } from 'lattice-ui-kit';
+import { ReduxUtils, useRequestState } from 'lattice-utils';
+import { useDispatch } from 'react-redux';
+import { RequestStates } from 'redux-reqseq';
+import type { UUID } from 'lattice';
+import type { RequestState } from 'redux-reqseq';
+
+import { ModalBody } from '../../../components';
+import { resetRequestState } from '../../../core/redux/actions';
+import { ORGANIZATIONS } from '../../../core/redux/constants';
+import { DELETE_EXISTING_ORGANIZATION, deleteExistingOrganization } from '../actions';
+
+const { isSuccess } = ReduxUtils;
+
+type Props = {
+  isOwner :boolean;
+  isVisible :boolean;
+  onClose :() => void;
+  organizationId :UUID;
+};
+
+const DeleteOrgModal = ({
+  isOwner,
+  isVisible,
+  onClose,
+  organizationId,
+} :Props) => {
+
+  const [standbyMessage, updateStandbyMessage] = useState(
+    'Are you sure you want to delete this organization? This action cannot be undone.'
+  );
+
+  const dispatch = useDispatch();
+
+  const deleteOrgRS :?RequestState = useRequestState([ORGANIZATIONS, DELETE_EXISTING_ORGANIZATION]);
+
+  const handleOnClickPrimary = () => {
+    if (isOwner) {
+      dispatch(deleteExistingOrganization(organizationId));
+    }
+    else {
+      updateStandbyMessage('You must be an owner to delete an organization.');
+    }
+  };
+
+  useEffect(() => {
+    if (isSuccess(deleteOrgRS)) onClose();
+  });
+
+  useEffect(() => () => dispatch(resetRequestState([DELETE_EXISTING_ORGANIZATION])));
```

Comment:
This will never run because the modal is never unmounted.

Reply:
oops you're right. i'd confused myself into thinking this works, because when the delete request is a success, the modal closes but also the page is redirected to the orgs list page, meaning that the modal does actually unmount. but if you just close the modal, you're right, the modal is not unmounted. anyhow, this is all solved with `ResetOnUnmount`, thank you! i'll remove this.